### PR TITLE
[SB-31] Webapp >  Sidebar fix

### PR DESCRIPTION
**Pull request by imariel2d**
**Things that i did**
- App.container had missing styles (it's a margin with the for the size of the sidebar)
- Indented scss files
- Added extra styles on sidebar (buttons outline)
- Refactored sidebar into component and container

**Evidence**
https://www.loom.com/share/7c47fbd5159c458e860c0acbe97c9760

**Jira**
https://songbox-io.atlassian.net/browse/SB-31?atlOrigin=eyJpIjoiYmY3NWJlM2Q4OWVmNDBlZGJlZWQxN2UwYmNkNzU5MzQiLCJwIjoiaiJ9

### CODE REVIEW

File: src/components/Sidebar/Sidebar.component.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import * as FaIcons from 'react-icons/fa';
+import * as AiIcons from 'react-icons/ai';
```
Comment (tonymtz) [inline src/components/Sidebar/Sidebar.component.js]: Import just the required icons.

File: src/components/Sidebar/Sidebar.component.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          </li>
+
+          <li className="profile-container">
+            <Profile />
+          </li>
+          {
+            SidebarData.map((item, index) => (
+              <Item
+                key={item.title}
+                index={index}
```
Comment (tonymtz) [inline src/components/Sidebar/Sidebar.component.js]: Why is this `index` needed? 🤔

File: src/components/Sidebar/Sidebar.container.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+const SidebarContainer = () => {
+  const nodeRef = React.useRef();
+
+  const [sidebar, setShowingSidebar] = useState(false);
+
+  useEffect(() => {
+    const clickOutside = (e) => {
+      if (nodeRef.current.contains(e.target)) {
+        return;
+      }
```
Comment (tonymtz) [inline src/components/Sidebar/Sidebar.container.jsx:14]: Nice

File: src/components/Sidebar/Sidebar.container.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+      e.stopPropagation();
+      setShowingSidebar(false);
+    };
+
+    document.addEventListener('mousedown', clickOutside, false);
+
+    return () => {
+      document.removeEventListener('mousedown', clickOutside, false);
```
Comment (tonymtz) [inline src/components/Sidebar/Sidebar.container.jsx:23]: Nice x 2

File: src/styles/app.scss [hunk 5]
```diff
@@ -1,65 +1,69 @@
...
+    pointer-events: none;
+  }
+
+  &-header {
+    align-items: center;
+    background-color: $dark-color;
+    color: $white;
     display: flex;
+    flex-direction: column;
+    font-size: calc(10px + 2vmin);
```
Comment (tonymtz) [inline src/styles/app.scss]: This is not a good practice.

**Review by tonymtz — APPROVED** [review]
Please fix the comments before merging this. The CSS looks wrong to me, but we will address this later.

File: src/components/Sidebar/Sidebar.component.js (hunk 2 above)
Comment (arielplas) [reply src/components/Sidebar/Sidebar.component.js]: It was to color the background of the "selected" route, but i found a better way to do it. I'll make a task on jira to remove the unnecessary selector ✨

**Pull request merged**